apps/files/lib/Service/OwnershipTransferService.php

```diff
@@ -388,14 +387,22 @@ private function collectIncomingShares(string $sourceUid,
 			if (empty($sharePage)) {
 				break;
 			}
-			if ($addKeys) {
-				foreach ($sharePage as $singleShare) {
-					$shares[$singleShare->getNodeId()] = $singleShare;
-				}
-			} else {
-				foreach ($sharePage as $singleShare) {
-					$shares[] = $singleShare;
-				}
+
+			if ($path !== null && $path !== "$sourceUid/files") {
+				$userFolder = $this->rootFolder->getUserFolder($sourceUid);
+				$normalizedPath = Filesystem::normalizePath($path);
+
+				$sharePage = array_filter($sharePage, static function (IShare $share) use ($userFolder, $normalizedPath) {
+					try {
+						return str_starts_with(Filesystem::normalizePath($userFolder->getFirstNodeById($share->getNodeId())->getPath() . '/', false), $normalizedPath . '/');
```

Comment:
```suggestion
						return str_starts_with(Filesystem::normalizePath($share->getTarget()) . '/', false), $normalizedPath . '/');
```
This should work, no?